py/picca/delta_extraction/masks/dla_mask.py

```diff
@@ -44,61 +46,55 @@ def dla_profile(lambda_, z_abs, nhi):
     DLA column density in log10(cm^-2)
     """
     transmission = np.exp(
-        -tau_lya(lambda_, z_abs, nhi)
-        -tau_lyb(lambda_, z_abs, nhi))
+        -compute_tau(lambda_, z_abs, nhi, LAMBDA_LYA, OSCILLATOR_STRENGTH_LYA, GAMMA_LYA)
+        -compute_tau(lambda_, z_abs, nhi, LAMBDA_LYB, OSCILLATOR_STRENGTH_LYB, GAMMA_LYB)
+    )
     return transmission
 
-### Implementation of Pasquier code,
-###     also in Rutten 2003 at 3.3.3
-LAMBDA_LYA = float(ABSORBER_IGM["LYA"]) ## Lya wavelength [A]
-@njit
-def tau_lya(lambda_, z_abs, nhi):
-    """Compute the optical depth for Lyman-alpha absorption.
+# constants to compute the optical depth of the DLA absoprtion
+ELEMENTARY_CHARGE = 1.6021e-19  # C
+EPSILON_0 = 8.8541e-12  # C^2.s^2.kg^-1.m^-3
+PROTON_MASS = 1.6726e-27  # kg
+ELECTRON_MASS = 9.109e-31  # kg
+BOLTZMAN_CONSTANT_K = 1.3806e-23  # m^2.kg.s^-2.K-1
+GAS_TEMP = 5 * 1e4  # K
 
-    Arguments
-    ---------
-    lambda_: array of floats
-    Wavelength (in Angs)
+# precomputed factors to save time
+# compared to equation 36 of Garnett et al 2017 there is a sqrt(2) missing
+# this is because otherwise we need to divide this quantity by sqrt(2)
+# when calling scipy.special.voigt
+GAUSSIAN_BROADENING_B = np.sqrt(BOLTZMAN_CONSTANT_K * GAS_TEMP / PROTON_MASS)
+# the 1e-10 appears as the wavelengths are given in Angstroms
+LORENTZIAN_BROADENING_GAMMA_PREFACTOR = 1e-10 / (4 * np.pi)
+TAU_PREFACTOR = (
+    ELEMENTARY_CHARGE**2 * 1e-10 / ELECTRON_MASS / SPEED_LIGHT / 4 / EPSILON_0)
 
-    z_abs: float
-    Redshift of the absorption
+def compute_tau(lambda_, z_abs, log_nhi, lambda_t, oscillator_strength_f, gamma):
+    r"""Compute the optical depth for Lyman-alpha absorption.
```

Comment:
This function doesn't seem specific to Lya

Reply:
I fixed the docstring